test/generators/stimulus_reflex_generator_test.rb

```diff
@@ -5,54 +5,48 @@
 require "./lib/generators/stimulus_reflex/stimulus_reflex_generator"
 
 class StimulusReflexGeneratorTest < Rails::Generators::TestCase
-  tests StimulusReflexGenerator
-  destination File.expand_path("../../tmp", __FILE__)
-  setup :prepare_destination
+  # tests StimulusReflexGenerator
+  # destination File.expand_path("../../tmp", __FILE__)
+  # setup :prepare_destination
 
-  test "creates singular named controller and reflex files" do
-    run_generator %w[demo]
-    assert_file "app/javascript/controllers/application_controller.js"
-    assert_file "app/javascript/controllers/demo_controller.js", /Demo/
-    assert_file "app/reflexes/application_reflex.rb"
-    assert_file "app/reflexes/demo_reflex.rb", /DemoReflex/
-  end
+  # test "creates singular named controller and reflex files" do
+  #   run_generator %w[demo]
+  #   assert_file "app/javascript/controllers/demo_controller.js", /Demo/
```

Comment:
I tried to investigate and the reason why the assertions fail is that [`assert_file`](https://api.rubyonrails.org/classes/Rails/Generators/Testing/Assertions.html#method-i-assert_file:~:text=given%20file%20exists.-,You%20need%20to%20supply%20an%20absolute%20path%20or%20a%20path%20relative%20to%20the%20configured%20destination,-%3A) expect an absolute path or a path relative to the configured destination.

So, in this case, we can use `../../app/javascript/controllers/demo_controller.js`.

However, the match assertions will still fail because the controllers are defined as anonymous classes:
```
import ApplicationController from './application_controller'

export default class extends ApplicationController {
  connect () {
    super.connect()
  }
}
```
So, we should just assert the presence of the file, I guess:
```suggestion
 assert_file "../../app/javascript/controllers/demo_controller.js"
```

Reply:
With the careful caveat that I am a junior level test comprehender, I think I have the test "working" again. You were right about the anonymous class matcher for the Stimulus classes, for sure.

In order to get this both a) working and b) not generating temporary test files in the gem's root folder, I had to modify the generator to behave differently in the test environment. This suggests that testing tasks which operate on `stdin` is hard, and there's probably a gem or technique which would allow us to obtain the same end result without blocking the thread. So I'll take full responsibility for demanding a UX that relies on such barbaric techniques.

However, part of the reason I subconsciously avoid programmatic tests is precisely that it seems like huge amounts of time and energy seem to get sucked into the toil of making tests pass, with very little accounting for actual coverage. After all, the worst bugs are the ones you cannot anticipate to test for.

Anyhow, I've always understood that you should try not to modify your source classes to make tests work, but I also recognize that having hard-coded paths makes things brittle in general, so I added the ternary to check for `Rails.env.test?` and we move on.

That said, it will take a fair bit of similar ternaries to get the main installer script into a testable state. It's not just the `stdin` prompts, but the entire system is writing files at (often) hard-coded locations. Leaving that challenge for another day.